src/public/modules/forms/admin/controllers/edit-logic-modal.client.controller.js

```diff
@@ -263,19 +268,38 @@ function EditLogicModalController(
 
     if (isNew) {
       vm.formLogics.push(vm.logic)
+      updateLogic({ form_logics: vm.formLogics }).then((error) => {
+        if (!error) {
+          $uibModalInstance.close()
+        }
+      })
       // Not new, and logic index is provided
     } else if (logicIndex !== -1) {
-      vm.formLogics[logicIndex] = vm.logic
+      vm.updateExistingLogic(logicIndex, vm.logic)
     }
-
-    updateLogic({ form_logics: vm.formLogics }).then((error) => {
-      if (!error) {
-        $uibModalInstance.close()
-      }
-    })
   }
 
   vm.cancel = function () {
     $uibModalInstance.close()
   }
+
+  vm.updateExistingLogic = function (logicIndex, updatedLogic) {
+    const logicIdToUpdate = vm.formLogics[logicIndex]._id
+    $q.when(
+      AdminFormService.updateFormLogic(
+        vm.myform._id,
+        logicIdToUpdate,
+        updatedLogic,
+      ),
+    )
+      .then(() => {
+        vm.formLogics[logicIndex] = updatedLogic
```

Comment:
`updatedLogic` should be the logic element returned by the backend

Reply:
edited